File: ui/src/models/entry.ts
import { ICategory } from "./category";

export interface IEntry {
    Id:string;
    DisplayName:string;
    Text:string;
    Keys:Array<string>;
    CategoryId:string;
    Category:ICategory;
} 

export class EntryConstructor implements IEntry {
    Id:string = "";
    DisplayName:string = "";
    Text:string = "";
    Keys:Array<string> = [];
    CategoryId:string = "";
    Category:ICategory = { Id:"", Name:"" };
}

File: backend/Lorebook.Convertor.Domain/Entry.cs
namespace Lorebook.Convertor.Domain;

public class Entry
{
    public Guid? Id { get; set; }
    public string? DisplayName { get; set; }
    public string? Text { get; set; }
    public IEnumerable<string> Keys { get; set; } = [];
    public Guid? CategoryId { get; set; }
    public Category? Category { get; set; }

    public static Entry From(LorebookEntry entry, IEnumerable<Category>? categories = null)
    {
        return new Entry
        {
            Id = entry.Id,
            DisplayName = entry.DisplayName,
            Text = entry.Text,
            Keys = entry.Keys,
            CategoryId = entry.Category,
            Category = categories?.FirstOrDefault(c => c.Id == entry.Category)
        };
    }
}

